Gems/GeoJSONSpawner/Code/Source/GeoJSONSpawner/GeoJSONSpawnerComponent.cpp

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (C) Robotec AI - All Rights Reserved
+ *
+ * This source code is protected under international copyright law.  All rights
+ * reserved and protected by the copyright holders.
+ * This file is confidential and only available to authorized individuals with the
+ * permission of the copyright holders.  If you encounter this file and do not have
+ * permission, please contact the copyright holders and delete this file.
+ */
+
+#include "GeoJSONSpawnerComponent.h"
+
+#include "AzCore/std/smart_ptr/make_shared.h"
+
+#include <AzCore/Asset/AssetSerializer.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzFramework/Physics/Common/PhysicsEvents.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+
+namespace GeoJSONSpawner
+{
+    GeoJSONSpawnerComponent::GeoJSONSpawnerComponent(
+        const AZStd::unordered_map<AZStd::string, GeoJSONUtils::GeoJSONSpawnableAssetConfiguration>& spawnableAssetConfigurations,
+        const AZStd::string& geoJsonFilePath,
+        AZ::u64 defaultSeed)
+        : m_spawnableAssetConfigurations(spawnableAssetConfigurations)
+        , m_geoJsonFilePath(geoJsonFilePath)
+        , m_defaultSeed(defaultSeed)
+    {
+    }
+
+    void GeoJSONSpawnerComponent::Activate()
+    {
+        AZ::TickBus::QueueFunction(
+            [this]()
+            {
+                SpawnEntities();
+            });
+        AZ::TickBus::Handler::BusConnect();
+        GeoJSONSpawnerRequestBus::Handler::BusConnect(GetEntityId());
+    }
+
+    void GeoJSONSpawnerComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+        GeoJSONSpawnerRequestBus::Handler::BusDisconnect();
+    }
+
+    void GeoJSONSpawnerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        GeoJSONUtils::GeoJSONSpawnableAssetConfiguration::Reflect(context);
+        GeoJSONUtils::GeoJSONSpawnableEntityInfo::Reflect(context);
+        GeoJSONUtils::GeometryObjectInfo::Reflect(context);
+
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<GeoJSONSpawnerComponent, AZ::Component>()
+                ->Version(0)
+                ->Field("SpawnableAssetConfigurations", &GeoJSONSpawnerComponent::m_spawnableAssetConfigurations)
+                ->Field("GeoJsonFilePath", &GeoJSONSpawnerComponent::m_geoJsonFilePath)
+                ->Field("DefaultSeed", &GeoJSONSpawnerComponent::m_defaultSeed);
+        }
+    }
+
+    void GeoJSONSpawnerComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        if (m_isInSpawningProcess)
```

Comment:
I am not sure how to use a QueueFunction in this case - if I understand QueueFunction correctly, it pushes a functor into the execution queue and the functor is called once on the nearest tick. But I need to know if the O3DE has despawned all the tickets that need to be despawned, which may take one tick or dozens of them. Is it possible to use QueueFunction like this without using something like the condition_variable?
For now, I can suggest an on-demand-tickbus-connect solution - when the spawner needs to despawn some of the tickets, it will connect to the TickBus. Then, when all tickets are despawned, OnTick calls the proper Spawn function and disconnects from the TickBus. This might be enough.

Reply:
I think logic that @patrykantosz proposed is Ok. I would consider to connect  TickBus when is needed and disconnect it when it is not needed. 
Let assume pseudo code:

```
Spawn()
AZ::TickBus::Handler::Connect()
```

And in OnTick
```
if (OnTickStuffDone())
{
AZ::TickBus::Handler::Disconnect()
}

```
It will eliminate this smelly m_isInModifyProcess ...